task/sql-tasks.js

```diff
@@ -280,7 +477,32 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        SELECT
+            DISTINCT Info.CompanyName,
+            Info.ProductName,
+            Info.UnitPrice as PricePerItem
+        FROM(
+            SELECT
+                CompanyName,
+                ProductName,
+                OrderDetails.UnitPrice
+            FROM Customers
+            JOIN Orders ON Customers.CustomerID = Orders.CustomerID
+            JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID
+            JOIN Products ON OrderDetails.ProductID = Products.ProductID
+        ) as Info
+        WHERE Info.UnitPrice = (
+            SELECT
+                MAX(OrderDetails.UnitPrice)
+            FROM Customers
+            JOIN Orders ON Customers.CustomerID = Orders.CustomerID
+            JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID
+            WHERE CompanyName = Info.CompanyName

```

Comment:
company name doesn't uniquely identify customer. also this column doesn't have an index so it's also slow.